new-frontiers-14/frontier-station-14
# Port Food Recipe Guidebook (EE#783)

## About the PR



Adds the Food Recipe guidebook page from [Einstein-Engines#783](https://github.com/Simple-Station/Einstein-Engines/pull/783) (Thank you, @Mnemotechnician!).  The Food Recipes page in the Guidebook now contains a dynamically populated list of recipes, taken from the in-game prototypes.

Made minor changes to support recipes with multiple output entities.

## Why / Balance


This feature will be very useful for #1935 - splitting recipes between a handful of machines means that it's easier to confuse what one recipe needs versus another.  Compare with bartending, where, worst case, you get an improperly mixed drink; in cooking, you can ruin individual ingredients.  Should improve quality of life overall, especially for newer players, or when introducing larger changes to cooking (cough cough, #1935).

## How to test


1. Open guidebook.
2. Check Food Recipes page.
3. Open recipe for cookies, result count should be 3 and not 1.

## Media


The guidebook page in use.  Shows an open recipe, the search bar in-use, and multiple filtered results.
![image](https://github.com/user-attachments/assets/20ec2017-b28f-49fd-a660-001fca2e66fc)

## Requirements



## Breaking changes


**Changelog**


:cl: Mnemotechnician
- The Food Recipes page in the guidebook now shows all available recipes.
